composeApp/src/androidMain/kotlin/it/fast4x/rimusic/utils/FileUtils.kt

```diff
@@ -34,6 +39,51 @@ fun copyDir(src: Path, dest: Path) {
     }
 }
 
+fun saveImageToInternalStorage(context: Context, imageUri: Uri, thumbnailName: String): Uri? {
+    try {
+        // Open input stream from the URI
+        val inputStream: InputStream? = context.contentResolver.openInputStream(imageUri)
+
+        // Create a new file in the app's internal storage
+        val outputFile = File(context.filesDir, thumbnailName)
+        val outputStream = FileOutputStream(outputFile)
+
+        // Copy the data from the input stream to the output stream (internal storage)
+        inputStream?.copyTo(outputStream)
+
+        // Close the streams
+        inputStream?.close()
+        outputStream.flush()
+        outputStream.close()
+
+        // Return the URI to the saved file in internal storage
+        return Uri.fromFile(outputFile)
+    } catch (e: IOException) {
+        Timber.e(e)
+        return null
+    }
+}
+
+fun checkFileExists(context: Context, filePath: String): String? {
+    val file = File(context.filesDir, filePath)
+
+    return if (file.exists()) {
+        file.toURI().toString()
+    } else {
+        null
+    }
+}
+
+fun deleteFileIfExists(context: Context, filePath: String): Boolean {
+    val file = File(context.filesDir, filePath)
+
+    return if (file.exists()) {
+        file.delete()
+    } else {
+        false
+    }
+}
+
```

Comment:
use `rememberLauncherForActivityResult`. Example [HERE](https://github.com/fast4x/RiMusic/blob/0419774a2b75a7cf0fda0d4f3ae45ded202cd5fc/composeApp/src/androidMain/kotlin/it/fast4x/rimusic/ui/screens/home/HomeSongsModern.kt#L356-L379)

Reply:
But these are operations on file that are stored in internal storage of the apps, so there is no need to ask user consent right? Hence, rememberLauncherForActivityResult is unnecessary right?